typescript/test/util.ts

```diff
@@ -0,0 +1,103 @@
+import * as assert from 'assert';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+
+const BASELINE_EXTENSION = '.baseline';
+export function compareToBaseline(outpurDir: string, baselineDir: string) {
+  // put all baseline files into fileStack
+  let fileStack: string[] = [];
+  const dirStack: string[] = [];
+  putAllBaselineFiles(baselineDir, fileStack, dirStack);
+  // store every item (file or directory with full path in output dir and baseline dir) in stack, pop once a time
+  const protoItemStack: Item[] = [];
+  putItemToStack(protoItemStack, outpurDir, baselineDir);
+  while (protoItemStack.length !== 0) {
+    const item = protoItemStack[0];
+    protoItemStack.shift();
+    // if item is a file, compare it with baseline
+    if (fs.lstatSync(item.outputPath).isFile()) {
+      const identical = checkIdenticalFile(
+        item.outputPath,
+        item.baselinePath + BASELINE_EXTENSION
+      );
+      // if two files are identical or it's generated properly, filter it from the stack.
+      if (identical !== 2) {
+        fileStack = fileStack.filter(
+          file => file !== item.baselinePath + BASELINE_EXTENSION
+        );
+      }
+    } else if (fs.lstatSync(item.outputPath).isDirectory()) {
+      // if item is a directory, loop the folder and put every item in stack again
+      putItemToStack(protoItemStack, item.outputPath, item.baselinePath);
+    }
+  }
+  if (fileStack.length !== 0) {
+    fileStack.forEach(file => {
+      console.error(file + ' is not identical with the generated file. ');
+    });
+  }
+  assert.strictEqual(fileStack.length, 0);
+}
+
+function checkIdenticalFile(
+  outputFullPath: string,
+  baselineFullPath: string
+): number {
+  if (!fs.existsSync(baselineFullPath)) {
+    console.error(baselineFullPath + 'is not generated. ');
```

Comment:
This case is not covered by an assertion.

Reply:
At least it seems so - please check the CircleCI logs of the recent run, the start script test should've failed because of the missing file but it passes. I'm not sure why the proto file is missing btw.

Reply:
yes, you are right, it should check whether `baselineFullPath ` exists or not. And after adding assertion to that, circleCI will fail coz it could not find `../../echo.proto.baseline` in baseline directory.

Reply:
And one more logic added, if it's a `.proto` file, we just skip it without comparing because baseline does not have protos, which makes tests all green now.